demos/_includes/DemoActionsUtil.php

```diff
@@ -131,7 +155,37 @@ public static function setupDemoActions(CountryLock $country): void
                 'args' => [
                     'age' => ['type' => 'integer', 'required' => true],
                     'city' => [],
-                    'gender' => ['type' => 'string', /* TODO 'values' => ['m' => 'Male', 'f' => 'Female'],*/ 'required' => true, 'default' => 'm'],
+                    'gender' => [
+                        'type' => 'string',
+                        'required' => true,
+                        'default' => 'm',
+                        'ui' => [
+                            'form' => [
+                                Dropdown::class, 'values' => ['m' => 'Male', 'f' => 'Female'],
+                            ],
+                        ],
+                    ],
+                ],
+                'fields' => [$country->fieldName()->iso3],
+                'callback' => function ($model, $age, $city, $gender) {
+                    $n = $gender === 'm' ? 'Mr.' : 'Mrs.';
+
+                    return 'Thank you ' . $n . ' at age ' . $age;
+                },
+                'preview' => function ($model, $age, $city, $gender) {
```

Comment:
why `$age`, ... params if model is passed as well?

I prefer to not pass the args as a) it is declaration order dependent I think b) it is not refactorable not testable with phpstan

same for `callback`

Reply:
$model here is the UserAction model and not the argument model. The arguments are send separately to preview callback.

Reply:
but with the same field names as args, right?

Reply:
No - in this case here $model is the Country model. 
I just push an update with proper type to each callback function. I will help clear the confusion. :)